precompile/fee_config_manager.go

```diff
@@ -0,0 +1,268 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/commontype"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	// must preserve order of these fields
+	gasLimitKey = iota
+	targetBlockRateKey
+	minBaseFeeKey
+	targetGasKey
+	baseFeeChangeDenominatorKey
+	minBlockGasCostKey
+	maxBlockGasCostKey
+	blockGasCostStepKey
+
+	minFeeConfigFieldKey = gasLimitKey
+	maxFeeConfigFieldKey = blockGasCostStepKey
+
+	numFeeConfigField = (maxFeeConfigFieldKey - minFeeConfigFieldKey + 1)
+)
+
+var (
+	_ StatefulPrecompileConfig = &FeeConfigManagerConfig{}
+
+	// Singleton StatefulPrecompiledContract for setting fee configs by permissioned callers.
+	FeeConfigManagerPrecompile StatefulPrecompiledContract = createFeeConfigManagerPrecompile(FeeConfigManagerAddress)
+
+	setFeeConfigSignature = CalculateFunctionSelector("setFeeConfig(uint256,uint256,uint256,uint256,uint256,uint256,uint256,uint256)")
+	getFeeConfigSignature = CalculateFunctionSelector("getFeeConfig()")
+
+	ErrCannotChangeFee = errors.New("non-enabled cannot change fee config")
+
+	// 8 fields in FeeConfig struct
+	feeConfigInputLen = common.HashLength * numFeeConfigField
+)
+
+// FeeConfigManagerConfig wraps [AllowListConfig] and uses it to implement the StatefulPrecompileConfig
+// interface while adding in the contract deployer specific precompile address.
+type FeeConfigManagerConfig struct {
+	AllowListConfig
+}
```

Comment:
Hmm I was thinking that we would include an updated fee config here, but this seems cleaner to just have the admin user send a transaction to update it.

Reply:
I was thinking that activation/actual fee config setting will be different actions. but with defaulting to chain configs and using a single-source of truth, actually we can add that here too. However I'm afraid that would introduce extra confusion since there will be 2 different fee configs in the genesis.